src/main/java/seedu/address/model/appointment/UniqueAppointmentList.java

```diff
@@ -47,23 +51,34 @@ public void add(Appointment toAdd) {
         internalList.add(toAdd);
     }
 
-    // /**
-    //  * Replaces the appointment {@code target} in the list with {@code editedAppointment}.
-    //  * {@code target} must exist in the list.
-    //  * The appointment identity of {@code editedAppointment} must not be the same as
-    //  * another existing appointment in the list.
-    //  */
-    // public void setAppointment(Appointment target, Appointment editedAppointment) {
-    //     // todo
-    // }
-    //
-    // /**
-    //  * Removes the equivalent appointment from the list.
-    //  * The appointment must exist in the list.
-    //  */
-    // public void remove(Appointment toRemove) {
-    //     // todo
-    // }
+    /**
+     * Replaces the appointment {@code target} in the list with {@code editedAppointment}.
+     * {@code target} must exist in the list.
+     * The appointment identity of {@code editedAppointment} must not be the same as
+     * another existing appointment in the list.
+     */
+    public void setAppointment(Appointment target, Appointment editedAppointment) {
+        requireAllNonNull(target, editedAppointment);
+        int index = internalList.indexOf(target);
+        if (index == -1) {
+            throw new AppointmentNotFoundException();
+        }
+        if (!target.isOverlapping(editedAppointment) && hasOverlaps(editedAppointment)) {
```

Comment:
Hmm perhaps the ! on line 66 should be omitted as it looks like it’s throwing an exception for duplicated appointments when it detects non overlapping appts instead. Let me know what you think about this 👍

Reply:
I think this is correct.
I believe that the check is if `target `and `editedAppointment `does not overlap, check to ensure that the `editedAppointment `does not overlap with any other appointments. 

If the `target `and `editedAppointment `do overlap, you are sure that there will be no problem adding the `editedAppointment` as it can safely replace the `target`.

Reply:
I agree with Jinhao. I think `!targer.isOverlapping` allows the appointment to be replaced without throwing `OverlappingAppoinmentException`, and the second condition is to check no other overlapping appointments exist in the appointment book.

Reply:
> Actually for simplicity, we can also consider this: call `remove` on target, then call `add` on `editedAppointment`. If it throws `OverlappingAppointmentException`, then we `add` back `target`
We should still throw a new exception when we `add` back the `target` right?

Reply:
Yes. It will still throw an exception.